src/Xabe.FFmpeg.Downloader/Android/AndroidFFmpegDownloader.cs

```diff
@@ -44,45 +46,24 @@ private string GenerateLink(OperatingSystemArchitecture arch)
             return string.Empty;
         }
 
-        public override async Task GetLatestVersion(string path)
+        public override async Task GetLatestVersion(string path, IProgress<float> progress = null)
         {
             OperatingSystemArchitecture arch = _operatingSystemArchitectureProvider.GetArchitecture();
 
-            await GetLatestVersionForArchitecture(path, arch);
+            await GetLatestVersionForArchitecture(path, arch, progress);
         }
 
-        protected async Task GetLatestVersionForArchitecture(string path, OperatingSystemArchitecture arch)
+        protected async Task GetLatestVersionForArchitecture(string path, OperatingSystemArchitecture arch, IProgress<float> progress = null)
         {
             if (!CheckIfFilesExist(path))
             {
                 return;
             }
 
             string link = GenerateLink(arch);
-            var fullPackZip = await DownloadFile(link, arch);
+            var fullPackZip = await DownloadFile(link, progress);
```

Comment:
You removed arch from this method so right know arch is not pass anywhere

Reply:
The architecture is not required there anymore since we generate the link before we call download. And I removed the override of DownloadFile and use the version in FFmpegDownloaderBase